runtime/signals_nat.c

```diff
@@ -188,9 +209,27 @@ extern void caml_stack_overflow(caml_domain_state*);
 CAMLno_asan
 DECLARE_SIGNAL_HANDLER(segv_handler)
 {
-  struct sigaction act;
   char * fault_addr;
 
+#if defined(INFO_CODE) && !defined(SYS_macosx)
+  /* If the signal was sent by a process, CONTEXT_FAULTING_ADDRESS is
+     not guaranteed to be valid and up-to-date. In this case, we make
+     sure to call the default handler by re-raising the exception.
+
+     On MacOS, one cannot reliably detect if the signal comes from a
+     process (http://www.openradar.me/FB8707529). However, on this
+     platform si_addr is valid and null in this case, so the rest of
+     the function is well-defined. This leads to another issue though:
+     the first time a SIGSEGV is sent to this process by another
+     process, execution continues with stack overflow detection turned
+     off. */
+  if (INFO_CODE == SI_USER || INFO_CODE == SI_QUEUE || INFO_CODE <= 0) {
```

Comment:
What's the INFO_CODE <= 0 case for? (The other two make sense to me)

Reply:
According to the link
> If si_code is SI_USER or SI_QUEUE, [XSI] [Option Start]  or any value less than or equal to 0, [Option End]  then the signal was generated by a process 

On Linux with pthreads linked, you get `SI_TKILL` (-4).

Reply:
And yes, I checked that the BSDs conform to this XSI extension.